src/components/CompanySelector.vue

```diff
@@ -0,0 +1,226 @@
+<template>
+  <div>
+    <div class="input">
+      <input
+        type="text"
+        placeholder="Select your company"
+        ref="searchCompanyInput"
+        v-model="searchCompanyInput"
+      >
+    </div>
+    <div
+      class="company-list"
+      :class="{ 
+        'company-list--loading': isLoading
+      }"
+    >
+      <div
+        v-if="items.length === 0 && searchCompanyInput"
+        class="company-list__item--noList"
+        v-text="'No result.'"
+      />
+      <div
+        v-else
+        v-for="item in items"
+        :key="item.objectID"
+        @click="selectCompany(item)"
+        class="company-list__item"
+        :class="{
+          'company-list__item--selected': (item.user_id === selectedCompanyId)
+        }"
+      >
+        <figure 
+          v-if="item.companyPhotoUrl"
+        >
+          <img :src="item.companyPhotoUrl" :alt="item.companyName">
+        </figure>
+        <figure
+          class="company-list__logo"
+          v-else
+        >
+          <img src="@/assets/user.svg" :alt="item.companyName">
+        </figure>
+        <p>{{ item.companyName }}</p>
+      </div>
+    </div>
+    <div v-if="!onlyInput">
+      <a href="#">Couldn't find your company?</a>
+      <button
+        class="button"
+        :class="{ 'button--signal': selectedCompany }"
+        @click="nextStep"
+      >
+        Next
+      </button>
+    </div>
+  </div>
+</template>
+
+<script>
+import API from "../api";
+
+export default {
+  props: ['onlyInput'],
+  name: "CompanySelector",
+  data() {
+    return {
+      items: {},
+      searchCompanyInput: '',
+      selectedCompany: null,
+      selectedCompanyId: null,
+      isLoading: false,
+    }
+  },
+  mounted() {
+    this.companySearch();
+  },
+  methods: {
+    debounce(func, wait, immediate) {
```

Comment:
It's great to have a `debounce` function. And, at the moment, I guess it is only used on this component. However, it seems to me that it does not depend on the component state. I would recommend moving it outside to a standalone function.

Reply:
https://github.com/anonymouslyreportingtool/webapp/pull/7/commits/2d295bb4d98895a8d248649409db7dcb87c476d4